sgnlp/base/preprocessor.py

```diff
@@ -0,0 +1,123 @@
+from abc import abstractmethod
+from typing import Tuple, Dict, Any, Sized, List
+
+import torch
+
+
+def simple_merge(batches: List[Dict[str, list]]):
+    keys = batches[0].keys()
+    merged = {}
+
+    for key in keys:
+        merged[key] = []
+        for batch in batches:
+            merged[key].extend(batch[key])
+
+    return merged
+
+
+def get_merge_function(
+    return_as_tensor=True,
+    padding=True,
+    pad_id=0,
+    pad_direction="right",
+    max_padding_lengths: Dict[str, int] = None,
+):
+    def merge_function(batches: List[Dict[str, list]]):
```

Comment:
I think it would be great to provide a small note here to explain what the merge function does so that end user will have some idea what to do if they choose to overload the merge function in the Preprocessor and/or if they are trying to debug or understand the preprocessor behaviour.

Reply:
Sure, I'll add some documentation in both the preprocessor base class and this function.